src/helpers/gcov.ts

```diff
@@ -0,0 +1,23 @@
+import glob from 'fast-glob'
+import { isProgramInstalled, runExternalProgram } from "./util"
+
+export function generateGcovCoverageFiles(projectRoot: string, include: string[] = [], ignore: string[] = [], gcovArgs = '',) {
+    if (!isProgramInstalled('gcov')) {
+        throw new Error('gcov is not installed, cannot process files')
+    }
+
+    const globstar = (pattern: string) => `**/${pattern}`
+    const gcovInclude = ['*.gcno', ...include].map(globstar)
+    const gcovIgnore = ['bower_components', 'node_modules', 'vendor', ...ignore].map(globstar)
+    const files = glob
+        .sync(gcovInclude, {
+            cwd: projectRoot,
+            ignore: gcovIgnore,
+            onlyFiles: true,
+        })
+    if (!files.length) {
+        throw new Error('No gcov files found')
+    }
+    const gcovLog = runExternalProgram('gcov', ['-pb', gcovArgs, files.join(' ')]);
```

Comment:
`gcovArgs` and `files` will each be passed in as one argument to gcov. Is this intended?

Reply:
Not sure what you mean here, they are passed into `runExternalProgram` using the `optionalArguments` field.

Reply:
```suggestion
    const gcovLog = runExternalProgram('gcov', ['-pb', gcovArgs, ...files]);
```

There may be nuance with the way `gcovArgs` is passed. While it's a string, it may actually be treated as _one_ string in the context of it being an argument to gcov. For example:

```
const gcovArgs = "-B main -d --verbose";
// %expected-type: string

gcovArgs;
// => "-B main -d --verbose"
```
```
const gcovArgs = "-B main -d --verbose".split(" ");
// %expected-type: string[]

gcovArgs[0];
// => "-B"
```